### PR TITLE
fix: Resolve the error of missing map file.

### PR DESCRIPTION
**Pull request by limerickgds**
**Description:** When there is no sourcemap file, an error occurs because the file path with .map added automatically cannot be found.

**BREAKING CHANGE:** no

**Related issue (if exists):**

### CODE REVIEW

**Comment by limerickgds** [comment]
@kdy1 Please help review this.

**Review by kdy1 — CHANGES_REQUESTED** [review]
Can you add a test for this?

**Comment by limerickgds** [comment]
> Can you add a test for this?

@kdy1 done

File: crates/swc/src/lib.rs [hunk 1]
```diff
@@ -324,43 +324,65 @@ impl Compiler {
...
-                                            "failed to read input source map from file at {}",
-                                            path
-                                        )
-                                    },
-                                )?))
+                                match map_path {
+                                    Some(map_path) => {
+                                        let path = map_path.display().to_string();
+                                        let file = File::open(&path)
+                                            .context("failed to open input source map file");
```
Comment (kdy1) [inline crates/swc/src/lib.rs:360]: This `context` is redundant

File: crates/swc/src/lib.rs [hunk 2]
```diff
@@ -324,43 +324,65 @@ impl Compiler {
...
-                                    None => {
-                                        format!("{}.map", dir.join(filename).display())
+                                // Old behavior. This check would prevent
+                                // regressions.
+                                // Perhaps it shouldn't be supported. Sometimes
+                                // developers don't want to expose their source code.
+                                // Map files are for internal troubleshooting
+                                // convenience.
+                                let fallback_map_path =
+                                    PathBuf::from(format!("{}.map", filename.display()));
```
Comment (kdy1) [inline crates/swc/src/lib.rs:334]: Can we be lazy here? This allocation is not necessary.

**Review by kdy1 — CHANGES_REQUESTED** [review]

**Review by kdy1 — APPROVED** [review]
swc-bump:
 - swc

**Pull request merged**